docs/content/preview/contribute/core-database/build-and-test.md

```diff
@@ -0,0 +1,251 @@
+headcontent: Building and testing with yb_build.sh.
+menu:
+  preview:
+    identifier: build-and-test
+    parent: core-database
+    weight: 2913
+type: docs
+---
+
+`yb_build.sh` is a bash script for both building and testing.
+Some flags can turn off certain parts of build, and other flags can decide which tests to run.
+In general, they come down to these components:
+
+- build:
+  - C++: C++ code in `src/yb/` and `ent/src/yb` for `yb-master`, `yb-tserver`, `yb-admin`, test binaries, etc.
+  - postgres: C code in `src/postgres/` for `postgres`, `ysql_dump`, `libyb_pgbackend`, `libpq`, etc.
+  - Java: Java code in `java/` for CDC, Java tests, drivers, etc.
+  - `initdb`: pre-created initial system catalog snapshot for YSQL
+- test:
+  - C++ tests
+  - Java tests
+
+For a full list of options, run `yb_build.sh -h`.
+What follows are some of the common options to pay attention to.
+
+## Build
+
+### Targets
+
+By default, all items are built.
+You can specify a target to narrow down what to build.
+See `./yb_build.sh -h` for the supported targets.
+Besides that, certain flags may skip building some items.
+For example, specifying a flag to run a C++ test skips the Java build (but running a Java test won't skip the C++ build).
+
+Although there is some intelligence to avoid rebuilding parts, it is incomplete.
+For example, the postgres build uses a build stamp calculated from a git commit and working changes; if there is a mismatch, it reruns postgres configure and make.
+Java is worse in that it runs all the build even if there are no changes since the last run.
+Use the `--skip-cxx-build`/`--scb` and `--skip-java-build`/`--sj` flags to reduce incremental build times.
+
+`initdb` is a special case because it is only built if it hasn't been built before, but it isn't rebuilt again unless the `reinitdb` target is specified.
+You should run `reinitdb` in case the initial system catalog in the source code is different since the last time `initdb` was run, unless you intentionally want to use the old initial system catalog (for example, to test upgrades).
+
+### Build generator
+
+CMake is used.
+
+If there are any changes to CMake files since the last build, the next incremental build may throw a build error, such as the following:
+
+```output
+FAILED: build.ninja
+```
+
+In that case, run the build with `--force-run-cmake`/`--frcm`.
+
+### Build tool
+
+By default, `ninja` is used, but `make` is also supported.
+Specify the build tool using the `--make` and `--ninja` flags.
+
+### Build type
+
+The following build types are available:
+
+- debug (the default)
+- release
+- fastdebug: fewer compiler optimizations and a few other tweaks on debug
+- asan: address sanitizer
+- tsan: thread sanitizer
+
+### Compiler
+
+By default, `clang` is used, but `gcc` is also supported.
+
+Specify the compiler using the `--gcc<version_number>` and `--clang<version_number>` flags.
+For example, `--gcc11` or `--clang15`.
```

Comment:
Perhaps don't specify examples with exact versions of GCC and Clang, so we don't have to update these examples.